src/app/auth/auth-direct-login/auth-direct-login.component.ts

```diff
@@ -59,27 +57,40 @@ export class AuthDirectLoginComponent implements OnInit {
 
   /**
    * Redirect user to a specific page if data is passed in, otherwise redirect to program switcher page
+   *
+   * @param {boolean}   redirectLater
+   * @returns {Promise<boolean | void>}
    */
   private async _redirect(redirectLater?: boolean): Promise<boolean | void> {
     const redirect = this.route.snapshot.paramMap.get('redirect');
-    const timelineId = +this.route.snapshot.paramMap.get('tl');
     const activityId = +this.route.snapshot.paramMap.get('act');
     const contextId = +this.route.snapshot.paramMap.get('ctxt');
     const assessmentId = +this.route.snapshot.paramMap.get('asmt');
     const submissionId = +this.route.snapshot.paramMap.get('sm');
     const topicId = +this.route.snapshot.paramMap.get('top');
+    let timelineId = +this.route.snapshot.paramMap.get('tl');
+
     // clear the cached data
     this.utils.clearCache();
+
+    // if timelineId "0", then try get cached timelineId
+    timelineId = (timelineId > 0) ? timelineId : this.storage.getUser().timelineId;
```

Comment:
@trtshen why do we need to get timeline id from storage?

Reply:
the "tl" must not be optional?

Reply:
@shawnm0705 the "tl" cannot be optional?